Sources/Sentry/SentryHttpTransport.m

```diff
@@ -133,7 +136,11 @@ - (void)sendEnvelope:(SentryEnvelope *)envelope
     // thread, which could be the main thread.
     __weak SentryHttpTransport *weakSelf = self;
     [self.dispatchQueue dispatchAsyncWithBlock:^{
-        [weakSelf.fileManager storeEnvelope:envelopeToStore];
+        NSString *path = [weakSelf.fileManager storeEnvelope:envelopeToStore];
+        if (![[NSFileManager defaultManager] fileExistsAtPath:path]) {
+            SENTRY_LOG_DEBUG(@"Could not store envelope. Schedule for sending.");
+            [weakSelf.notStoredEnvelopes addObject:envelopeToStore];
```

Comment:
`h`: We will run into threading issues here. `sendAllCachedEnvelopes` can be called from multiple threads, but it's thread-safe by ensuring it runs only once. Still, it can happen that `sendAllCachedEnvelopes` removes an object from `notStoredEnvelopes` on a different thread while we add an object here. To solve this, I suggest adding `@synchronized` when accessing `notStoredEnvelopes`. The extra lock is acceptable as it's an edge case and it's executed on a bg thread.